compiler/shared/test/diff/mono.mls

```diff
@@ -0,0 +1,1259 @@
+//│ Lifted:
+//│ TypingUnit {Code(List(if ('(' b ')') then 1 else 2))}
+//│ Mono:
+//│ 
+//│ Defunc result: 
+//│ main$$0()
+//│ fun main$$0() =
+//│   if b then #1 else #2
+//│ 1 | 2
+//│ res
+//│     = 2
+
+:mono
+((f, g) => f(g))(f => f, true)
+//│ Parsed:
+//│ TypingUnit(App(Bra(rcd = false, Lam(Tup(_: Var(f), _: Var(g)), App(Var(f), Tup(_: Var(g))))), Tup(_: Lam(Tup(_: Var(f)), Var(f)), _: Var(true))))
+//│ Lifted:
+//│ TypingUnit {
+//│   class Lambda2$1$1([]) {fun apply = (f, g,) => f(g,)}
+//│   class Lambda1$2$2([]) {fun apply = (f,) => f}
+//│   Code(List('(' {new Lambda2$1$1([]) {}} ')'({new Lambda1$2$2([]) {}}, true,)))
+//│ }
+//│ Mono:
+//│ 
+//│ Defunc result: 
+//│ main$$2()
+//│ fun apply$Lambda2$1$1(this, f, g) =
+//│   f match {case obj: Lambda1$2$2 => apply$Lambda1$2$2(obj, g)}
+//│ fun main$$2() =
+//│   new Lambda2$1$1 ()  match {case obj: Lambda2$1$1 => apply$Lambda2$1$1(obj, new Lambda1$2$2 () , true)}
+//│ fun apply$Lambda1$2$2(this, f) =
+//│   f
+//│ class Lambda2$1$1() {
+//│ }
+//│ class Lambda1$2$2() {
+//│ }
+//│ true
+//│ res
+//│     = true
+
+
+:mono
+(b => if b then true else false) (true)
+//│ Parsed:
+//│ TypingUnit(App(Bra(rcd = false, Lam(Tup(_: Var(b)), If(IfThen(Var(b), Var(true), Some(Var(false))))), Tup(_: Var(true))))
+//│ Lifted:
+//│ TypingUnit {
+//│   class Lambda1$1$1([]) {fun apply = (b,) => if (b) then true else false}
+//│   Code(List('(' {new Lambda1$1$1([]) {}} ')'(true,)))
+//│ }
+//│ Mono:
+//│ 
+//│ Defunc result: 
+//│ main$$1()
+//│ fun apply$Lambda1$1$1(this, b) =
+//│   if b then true else false
+//│ fun main$$1() =
+//│   new Lambda1$1$1 ()  match {case obj: Lambda1$1$1 => apply$Lambda1$1$1(obj, true)}
+//│ class Lambda1$1$1() {
+//│ }
+//│ Bool
+//│ res
+//│     = true
+
+:mono
+fun f(x) =
+  if(x > 0) then x+1 else x - 1
+f(2)+3
+//│ Parsed:
+//│ TypingUnit(NuFunDef(None, f, None, [], Lam(Tup(_: Var(x)), Blk(...))), App(Var(+), Tup(_: App(Var(f), Tup(_: IntLit(2))), _: IntLit(3))))
+//│ Lifted:
+//│ TypingUnit {
+//│   fun f$1 = (x,) => {if ('(' >(x, 0,) ')') then +(x, 1,) else -(x, 1,)}
+//│   Code(List(+(f$1(2,), 3,)))
+//│ }
+//│ Mono:
+//│ 
+//│ Defunc result: 
+//│ main$$1()
+//│ fun f$1(x) =
+//│   if >(x, #0) then +(x, #1) else -(x, #1)
+//│ fun main$$1() =
+//│   +(f$1(#2), #3)
+//│ fun f: Int -> Int
+//│ Int
+//│ res
+//│     = 6
+
+:mono
+fun fac(n) = 
+  if (n > 1) then fac(n - 1) * n else 1
+fac(2)
+//│ Parsed:
+//│ TypingUnit(NuFunDef(None, fac, None, [], Lam(Tup(_: Var(n)), Blk(...))), App(Var(fac), Tup(_: IntLit(2))))
+//│ Lifted:
+//│ TypingUnit {
+//│   fun fac$1 = (n,) => {if ('(' >(n, 1,) ')') then *(fac$1(-(n, 1,),), n,) else 1}
+//│   Code(List(fac$1(2,)))
+//│ }
+//│ Mono:
+//│ 
+//│ Defunc result: 
+//│ main$$1()
+//│ fun fac$1(n) =
+//│   if >(n, #1) then *(fac$1(-(n, #1)), n) else #1
+//│ fun main$$1() =
+//│   fac$1(#2)
+//│ fun fac: Int -> Int
+//│ Int
+//│ res
+//│     = 2
+
+:mono
+:e // FIXME: Malformed Test?
```

Comment:
@yuankeyu could you comment on this and propose a fixa? What were you trying to do?